### ISSUE TITLE
Cannot use hardCopyUsing when the original LinkedList is not empty

### ISSUE THREAD

**Issue by ekinkaradag** [ekinkaradag/data-structures#6]
If the original LinkedList is not empty, and we try to copy&overwrite another LinkedList to it like this;
```python
customList_one = LinkedList([1, 2, 3])
customList_two = LinkedList(["Orange", "Banana", "Apple", "Banana"])

customList_one.hardCopyUsing(customList_two)
```
we get an exception: `AttributeError: 'NoneType' object has no attribute 'data'`

**Comment by ekinkaradag** [comment]
### The reason behind this:
`remove` method is used in `hardCopyUsing()`, and `remove` deletes every element from the LinkedList. However, `remove` **tries** to do that by connecting the LinkedList's `head` to the next element in line. 

In the deletion process of the last element, `remove` tries to connect the LinkedList's `head` to the next element in line which actually does not even exist. That causes the exception.

This problem is planned to be solved with [PR#5](https://github.com/ekinkaradag/data-structures/pull/5)

**Comment by ekinkaradag** [comment]
The issue is fixed with #5
